### PR TITLE
[Bug] Fix issue with WRW heuristics instance selection

### PR DESCRIPTION
**Pull request by BrianHarrisonAMD**
After the fix for #3249 WRW heuristics would fail to parse split_k properly, and this would lead to using the default instance in all cases. This change will prevent issues where the split_k value is missing or malformed, and also allow for valid cases to parse.

### CODE REVIEW

File: test/gtest/unit_implicitgemm_ck_util.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+};
+} // namespace unit_implicitgemm_ck_util_test
+
+using namespace unit_implicitgemm_ck_util_test;
+
+struct CPU_UnitTestImplicitGemmCKUtil_NONE : CKArgParsingTest<StubbedWRWCKArgs, StubbedDeviceOps>
+{
+};
+
+// TEST_F(CPU_UnitTestImplicitGemmCKUtilWRW_NONE, TestParsing) { TestParsing(); }
```
Comment (BrianHarrisonAMD) [inline test/gtest/unit_implicitgemm_ck_util.cpp]: whoops ill remove

**Review by cderb — APPROVED** [review]
lgtm

File: src/include/miopen/solver/implicitgemm_ck_util.hpp [hunk 2]
```diff
@@ -137,10 +138,28 @@ bool IsCKArgsSupported(const ProblemDescriptionType& problem, const std::string&
...
+            {
+                split_k = std::stoi(kernel_id.substr(pos + 1));
+            }
+            catch(std::exception& e)
+            {
+                MIOPEN_LOG_WE("Unable to parse split_k from kernel_id for wrw: "
+                              << kernel_id << " : " << e.what());
+                return false;
+            }
+
```
Comment (CAHEK7) [inline src/include/miopen/solver/implicitgemm_ck_util.hpp:162]: Doesn't is complain "set but not used variable" on `split_k`?

**Review by CAHEK7 — APPROVED** [review]

**Comment by CAHEK7** [comment]
Probably it's better to use `Mock` instead of `Stubbed`, at least it's kind of widely used test terminology for fake instances of anything.

File: src/include/miopen/solver/implicitgemm_ck_util.hpp (hunk 2 above)
Comment (BrianHarrisonAMD) [reply src/include/miopen/solver/implicitgemm_ck_util.hpp:162]: It doesn't actually, but that is a bit surprising.

**Comment by BrianHarrisonAMD** [comment]
> Probably it's better to use `Mock` instead of `Stubbed`, at least it's kind of widely used test terminology for fake instances of anything.

Good point!
 I should do this with google mock, and the name should be Mock.

**Pull request merged**